File: p1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApplication9
{
    class Program
    {   

        static int TriboRecursivo(int n)
        {
            if (n == 0 ) return 0;
            if (n == 1 ) return 0;
            if (n == 2 ) return 1;
                return TriboRecursivo(n - 1) + TriboRecursivo(n - 2) +TriboRecursivo(n - 3);
        }


        static int TriboNormal(int n)
        {
            int a = -1, b = 1, c = 0, d = 0;
            for (int i = 1; i < n; i++)
            {
                a = b;
                b = c;
                c = d;
                d = a + b + c;
            }
            return d;
        }
        

        static void printTribRec(int n)
        {
            for (int i = 1; i <=n; i++)
                Console.Write(TriboRecursivo(i)+ ", ");
        }



        static void printTribNorm(int n)
        {
            for (int i = 1; i <= n; i++)
                Console.Write(TriboRecursivo(i) + ", ");
        }



        static void Main(string[] args)
        {
            int n;
            System.Console.WriteLine("====Tribonacci Estructurado====");
            System.Console.WriteLine("Introduce un numero n: ");
            n = Convert.ToInt32(System.Console.ReadLine());
            printTribNorm(n);

            System.Console.WriteLine();
            System.Console.WriteLine();
            
            System.Console.WriteLine("====Tribonacci Recursivo====");
            System.Console.WriteLine("Introduce un numero n: ");
            n = Convert.ToInt32(System.Console.ReadLine());
            printTribRec(n);
            System.Console.ReadKey();        
        }
    }
}

File: p1.fs
open System
let tribonormal n =
  let mutable a=(-1)
  let mutable b=1
  let mutable c=0
  let mutable d=0
  for i in 1..(n-1) do
    a<-b
    b<-c
    c<-d
    d<-a+b+c        
  d            
    

let rec triboRecursivo n =
    match n with
    | 0 -> 0
    | 1 ->0
    | 2 ->1
    | n ->triboRecursivo(n-1) + triboRecursivo(n-2)+triboRecursivo(n-3)

let tribRecursivo n=
     for i in 1..n do
        printf "%d, " (triboRecursivo i)

let tribNormal n=
     for i in 1..n do
        printf "%d, " (tribonormal i)


[<EntryPoint>]
let main argv = 
    //printfn "%A" argv

    // TRIBONACCI NORMAL
  Console.WriteLine("Introduzca un numero n:")
  let num1=Console.ReadLine()
  let n=Convert.ToInt32(num1)
    
  Console.WriteLine(tribNormal n)


    // TRIBONACCI RECURSIVO
  Console.WriteLine("Introduzca un numero n: ")
  let num1=Console.ReadLine()
  let n=Convert.ToInt32(num1)
  Console.WriteLine(tribRecursivo n)

  let tecla= Console.ReadKey()
  0 // devolver un código de salida entero
